subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/report/DefaultPerformanceFlakinessDataProvider.java

```diff
@@ -39,4 +43,25 @@ public BigDecimal getFlakinessRate(String scenario) {
     public BigDecimal getFailureThreshold(String scenario) {
         return failureThresholds.get(scenario);
     }
+
+    @Override
+    public ScenarioRegressionResult getScenarioRegressionResult(String scenarioName, double regressionPercentage) {
```

Comment:
How about moving this method to `PerformanceFlakinessDataProvider` and make it a `default` interface method?

Reply:
What would be the advantage?

Reply:
1. We don't need to write two `@Override public ScenarioRegressionResult getScenarioRegressionResult`
2. This can move `getScenarioRegressionResult`/`isBigRegression`/`isStableScenario` seems to belong to `PerformanceFlakinessDataProvider`, not a specific implementation class.

WDYT?

Reply:
I actually prefer not having `isBigRegression` / `isStableScenario ` in the interface, since the enum captures that already.

As a next step, I would like to distinguish between the raw result from the test run and the merged scenario. Currently, both is `ScenarioBuildResultData`. The merged result should have an enum as the status and the status would be decided on merging. Then the PerformanceFlakinessDataProvider would only be used there and not anywhere else and the index generator can focus on displaying and wouldn't need to access the raw data any more.